plugins/woocommerce/includes/class-wc-product-variation.php

```diff
@@ -583,4 +584,44 @@ protected function get_valid_tax_classes() {
 
 		return $valid_classes;
 	}
+
+	/**
+	 * Get the value of the "Cost of Goods Sold value overrides parent value" flag for this product.
+	 *
+	 * If the flag is set to true, the effective value is equal to the defined value for this variation.
+	 * Otherwise, the effective value is equal to the sum of the defined values for the variation and the parent product.
+	 *
+	 * @return bool The current value of the flag.
+	 */
+	public function get_cogs_value_overrides_parent(): bool {
+		return (bool) $this->get_prop( 'cogs_value_overrides_parent' );
+	}
+
+	/**
+	 * Set the value of the "Cost of Goods Sold value overrides parent value" flag for this product.
+	 *
+	 * WARNING! If the Cost of Goods Sold feature is disabled this value will NOT be persisted when the product is saved.
+	 *
+	 * @param bool $value The value to set for the flag.
+	 */
+	public function set_cogs_value_overrides_parent( bool $value ): void {
+		$this->set_prop( 'cogs_value_overrides_parent', $value );
+	}
+
+	/**
+	 * Get the effective value of the Cost of Goods Sold for this product.
+	 * (the final, actual monetary value).
+	 *
+	 * See get_cogs_value_overrides_parent.
+	 *
+	 * @return float
+	 */
+	public function get_cogs_effective_value(): float {
```

Comment:
I prefer this to be explicit because other types of COGS-related data/methods will be added in the future, e.g. `cogs_category`, `cogs_quantity_type` (like "fixed" or "percent") or `cogs_value_type` (like "materials" or "manufacturing").

Reply:
Makes sense 👍🏼